README.md

```diff
@@ -1,12 +1,42 @@
 # terraform-provider-infoblox
 Infoblox Provider for Terraform
 
-## Prerequisite 
+## Requirments
 
-* go
-* golang/dep
-* terraform
+* [Terraform](https://www.terraform.io/downloads.html) 
+* [Go](https://golang.org/doc/install) 1.9(to build the provider plugin)
+* [dep](https://github.com/golang/dep)
 
-## How to compile and build plugin
+## Building the Provider
 
-Just run *make build*
+clone repository to ```$GOPATH/src/github.com/infobloxopen/terraform-provider-infoblox```
+```
+$ mkdir -p $GOPATH/src/github.com/infobloxopen; cd $GOPATH/src/github.com/infobloxopen
+$ git clone git@github.com:infobloxopen/terraform-provider-infoblox
+```
+Enter the provider directory and build the provider
+```
+$ cd $GOPATH/src/github.com/infobloxopen/terraform-provider-infoblox
+$ make build
+```
+
+## Developing the Provider
+If you wish to work on the provider, you'll first need Go installed on your machine (version 1.9+ is required). You'll also need to correctly setup a GOPATH, as well as adding ```$GOPATH/bin``` to your ```$PATH```.
+
+To compile the provider, run ```make build```. This will build the provider and put the provider binary in the ```$GOPATH/bin``` directory.
+```
+$ make build
+...
+$ $GOPATH/bin/terraform-provider-infoblox
+...
+```
+In order to test the provider, you can simply run make test.
```

Comment:
Testing is also part of Developing

Reply:
https://github.com/terraform-providers/terraform-provider-google/blob/master/README.md
https://github.com/terraform-providers/terraform-provider-aws/blob/master/README.md

@jkraj  i took these two as references to do the README.md